AEPAssurance/Source/AssuranceSessionOrchestrator.swift

```diff
@@ -114,48 +140,75 @@ class AssuranceSessionOrchestrator: AssurancePresentationDelegate {
 
         /// display the error if the pin is empty
         if pin.isEmpty {
-            session.presentation.sessionConnectionError(error: .noPincode)
+            authorizingPresentation?.sessionConnectionError(error: .noPincode)
             terminateSession()
             return
         }
 
         /// display error if the OrgID is missing.
         guard let orgID = stateManager.getURLEncodedOrgID() else {
-            session.presentation.sessionConnectionError(error: .noOrgId)
+            authorizingPresentation?.sessionConnectionError(error: .noOrgId)
             terminateSession()
             return
         }
-
+        
+        authorizingPresentation?.sessionConnecting()
         Log.trace(label: AssuranceConstants.LOG_TAG, "Connect Button clicked. Starting a socket connection.")
         session.sessionDetails.authenticate(withPIN: pin, andOrgID: orgID)
         session.startSession()
     }
-    #if DEBUG
-    func quickConnectClicked(sessionID: String, environment: String, token: Int) {
-        if session != nil {
-            Log.warning(label: AssuranceConstants.LOG_TAG, "Quick connect attempted when active session exists")
-            return
-        }
-
-        // TODO: - Expose token in assurance details object
-        let assuranceDetails = AssuranceSessionDetails(sessionId: sessionID, clientId: stateManager.clientID, environment: AssuranceEnvironment.prod)
-        createSession(withDetails: assuranceDetails)
-    }
-    #endif
-    ///
-    /// Invoked when Cancel button is clicked on the PinCode screen.
-    ///
+    
     func pinScreenCancelClicked() {
         Log.trace(label: AssuranceConstants.LOG_TAG, "Cancel clicked. Terminating session and dismissing the PinCode Screen.")
         terminateSession()
     }
 
-    ///
-    /// Invoked when Disconnect button is clicked on the Status UI.
-    ///
     func disconnectClicked() {
         Log.trace(label: AssuranceConstants.LOG_TAG, "Disconnect clicked. Terminating session.")
         terminateSession()
     }
 
+    var isConnected: Bool {
+        get {
+            return session?.socket.socketState == .open
+        }
+    }
+    
+#if DEBUG
+    
+    func quickConnectBegin() {
```

Comment:
Would also be nice to move all of this code into an extension that contains the quickconnect stuff and the DEBUG flag stuff.

Reply:
The reason I removed the extensions is because when creating the mock for the AssuranceSessionOrchestrator you can only override @objc functions and types which meant that we would need to make the AssuranceConnectionError objc compatible. This would lead to losing the associated values in the enum. I thought the trade offs were worse in keeping the extension but we can go that direction if you disagree.

Reply:
Ahh, makes sense!  I retract my previous point -- I didn't take the mock object into account.